src/app/modules/convai-checker/convai-checker.component.ts

```diff
@@ -389,34 +398,44 @@ export class ConvaiChecker implements OnInit, OnChanges {
     if (this.demoSettings.modelName) {
       analyzeCommentData.modelName = this.demoSettings.modelName;
     }
-    this.mostRecentRequestSubscription =
-      this.analyzeApiService.checkText(
-          analyzeCommentData,
-          this.demoSettings.useGapi /* makeDirectApiCall */,
-          this.demoSettings.usePluginEndpoint ? this.pluginEndpointUrl : this.serverUrl)
-        .pipe(finalize(() => {
-          console.log('Request done');
-          let newScore = this.getMaxScore(this.analyzeCommentResponse);
-          this.statusWidget.notifyScoreChange(newScore);
-          this.scoreChanged.emit(newScore);
-          this.mostRecentRequestSubscription = null;
-        }))
-        .subscribe(
-          (response: AnalyzeCommentResponse) => {
-            this.analyzeCommentResponse = response;
-            this.analyzeCommentResponseChanged.emit(this.analyzeCommentResponse);
-            console.log(this.analyzeCommentResponse);
-            this.checkInProgress = false;
-            this.canAcceptFeedback = true;
-          },
-          (error) => {
-            console.error('Error', error);
-            this.checkInProgress = false;
-            this.canAcceptFeedback = false;
-            this.analyzeErrorMessage = this._getErrorMessage(error);
-            this.analyzeCommentResponse = null;
-          }
-        );
+
+    this.analyzeApiService.checkText(
+        analyzeCommentData,
+        this.demoSettings.useGapi /* makeDirectApiCall */,
+        this.demoSettings.usePluginEndpoint ? this.pluginEndpointUrl : this.serverUrl)
+      .pipe(finalize(() => {
+        console.log('Request done');
+        // TODO: Add a comment to explain why/how this.analyzeCommentResponse is set
```

Comment:
Regarding the TODOs:

The flow here is confusing, but finalize() actually happens AFTER the callbacks in subscribe/error despite appearing first sequentially in the code. Since finalize() happens regardless of whether the request succeeded or not, I don't think we should emit the response there, since it might not be from the most recent request. Unless I'm misunderstanding what your suggestion is.

I will follow up on looking into async-ness of notifyScoreChange tomorrow. I don't want to block on that.

Reply:
Thanks! I'll update the comment.  :)

Reply:
Actually, I just created a var for it, and re-ordered so it's more natural to read.

Reply:
SGTM!